### PR TITLE
MINOR: remove duplicate code of serializing auto-generated data

### PR DESCRIPTION
**Pull request by chia7712**
as title

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/SubscriptionInfo.java [hunk 1]
```diff
@@ -265,14 +265,7 @@ public ByteBuffer encode() {
                 "Should never try to encode a SubscriptionInfo with version [" +
                     data.version() + "] > LATEST_SUPPORTED_VERSION [" + LATEST_SUPPORTED_VERSION + "]"
             );
-        } else {
-            final ObjectSerializationCache cache = new ObjectSerializationCache();
-            final ByteBuffer buffer = ByteBuffer.allocate(data.size(cache, (short) data.version()));
-            final ByteBufferAccessor accessor = new ByteBufferAccessor(buffer);
-            data.write(accessor, cache, (short) data.version());
-            buffer.rewind();
```
Comment (dajac) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/SubscriptionInfo.java:273]: `MessageUtil.toByteBuffer` uses `flip` whereas `rewind` is used here. Does it make any difference in this particular case?

Comment (chia7712) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/assignment/SubscriptionInfo.java:273]: In this case, the state of ```flip``` is coincide with ```rewind```. This buffer is filled so the ```limit``` is always equal to ```capability``` even though ```rewind``` does not reset ```limit```.

**Review by dajac — APPROVED** [review]
LGTM

**Comment by chia7712** [comment]
the flay is traced by https://issues.apache.org/jira/browse/KAFKA-12245

**Pull request merged**